### PR TITLE
manifest: Update nrf_modem library and add nRF92 support

### PR DESCRIPTION
**Pull request by anhmolt**
See CHANGELOG file in nrfxlib for details about library features and changes.

### CODE REVIEW

File: lib/nrf_modem_lib/nrf_modem_lib.c [hunk 1]
```diff
@@ -69,6 +71,27 @@ static const struct nrf_modem_bootloader_init_params bootloader_init_params = {
...
 	.fault_handler = nrf_modem_fault_handler
 };
+#endif /* CONFIG_SOC_SERIES_NRF91X */
+
+#ifdef CONFIG_SOC_SERIES_NRF92X
+
+static const struct nrf_modem_init_params init_params = {
+	.shmem.ctrl = {
+		.base = DT_REG_ADDR(DT_NODELABEL(cpuapp_cpucell_ipc_shm)),
+		.size = CONFIG_NRF_MODEM_LIB_SHMEM_CTRL_SIZE,
```
Comment (eivindj-nordic) [inline lib/nrf_modem_lib/nrf_modem_lib.c:81]: Could we use DT for size as well and get rid of the KConfig values for nRF92?

File: lib/nrf_modem_lib/nrf_modem_os.c [hunk 2]
```diff
@@ -12,14 +12,21 @@
...
 #include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(nrf_modem, CONFIG_NRF_MODEM_LIB_LOG_LEVEL);
+
+#if CONFIG_SOC_SERIES_NRF91X
+#include <pm_config.h>
+#define SHMEM_TX_HEAP_ADDR (PM_NRF_MODEM_LIB_TX_ADDRESS)
+#elif CONFIG_SOC_SERIES_NRF92X
+#define SHMEM_TX_HEAP_ADDR (DT_REG_ADDR(DT_NODELABEL(cpuapp_cpucell_ipc_shm_heap)))
+#endif
+#define SHMEM_TX_HEAP_SIZE (CONFIG_NRF_MODEM_LIB_SHMEM_TX_SIZE)
```
Comment (eivindj-nordic) [inline lib/nrf_modem_lib/nrf_modem_os.c:25]: Should the size be given by DT for nRF92?

**Review by lemrey — APPROVED** [review]

**Review by divipillai — APPROVED** [review]

File: lib/nrf_modem_lib/nrf_modem_os_rpc.c [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+#include <zephyr/cache.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/mbox.h>
+#include <zephyr/ipc/icmsg.h>
+#include <zephyr/ipc/pbuf.h>
+
+#define DCACHE_LINE_SIZE 0
+
+struct nrf_modem_pbuf {
```
[the remark this answers is not in the document]
Comment (anhmolt) [reply lib/nrf_modem_lib/nrf_modem_os_rpc.c:24]: Thanks! I have added a short description for the different structs.

**Review by SeppoTakalo — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -763,9 +764,11 @@ Modem libraries
 
 * :ref:`nrf_modem_lib_readme`:
 
+  * Added support for socket option ``SO_IPV6_DELAYED_ADDR_REFRESH``.
   * Updated the RTT trace backend to allocate the RTT channel at boot, instead of when the modem is activated.
   * Rename the nRF91 socket offload layer from ``nrf91_sockets`` to ``nrf9x_sockets`` to reflect that the offload layer is not exclusive to the nRF91 Series SiPs.
   * Removed support for deprecated RAI socket options ``SO_RAI_LAST``, ``SO_RAI_NO_DATA``, ``SO_RAI_ONE_RESP``, ``SO_RAI_ONGOING``, and ``SO_RAI_WAIT_MORE``.
+  * Removed the mutex in the :c:func:`nrf9x_socket_offload_getaddrinfo` function after updating the :c:func:`nrf_getaddrinfo` function to handle concurrent requests.
```
Comment (divipillai) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:771]:
```suggestion
  * Rename the nRF91 socket offload layer from ``nrf91_sockets`` to ``nrf9x_sockets`` to reflect that the offload layer is not exclusive to the nRF91 Series SiPs.
  
  * Removed: 
    
    * Support for deprecated RAI socket options ``SO_RAI_LAST``, ``SO_RAI_NO_DATA``, ``SO_RAI_ONE_RESP``, ``SO_RAI_ONGOING``, and ``SO_RAI_WAIT_MORE``.
    * The mutex in the :c:func:`nrf9x_socket_offload_getaddrinfo` function after updating the :c:func:`nrf_getaddrinfo` function to handle concurrent requests.
```

File: lib/nrf_modem_lib/nrf_modem_lib.c (hunk 1 above)
Comment (anhmolt) [reply lib/nrf_modem_lib/nrf_modem_lib.c:81]: No, not currently. It must be set to the value of `CONFIG_NRF_MODEM_SHMEM_CTRL_SIZE` exported by `nrf_modem`. This is not in the devicetree. Let's see if we can improve this in some later work.

File: lib/nrf_modem_lib/nrf_modem_os.c (hunk 2 above)
Comment (anhmolt) [reply lib/nrf_modem_lib/nrf_modem_os.c]: Yes, I have updated this.

**Review by nordicjm — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by MarkusLassila — APPROVED** [review]

**Review by tokangas — APPROVED** [review]

**Comment by MarkusLassila** [comment]
Triggered re-running of failed tests. Serial LTE Modem tests should pass. Contabo server was restarted yesterday due to performance reasons. When it is at maximum capacity we start to see these DTLS errors with SLM tests.

**Review by juhaylinen — APPROVED** [review]

**Review by trantanen — APPROVED** [review]

File: tests/lib/date_time/CMakeLists.txt [hunk 5]
```diff
@@ -14,7 +14,8 @@ test_runner_generate(src/main.c)
 
 cmock_handle(${ZEPHYR_NRFXLIB_MODULE_DIR}/nrf_modem/include/nrf_modem_at.h
 	     FUNC_EXCLUDE ".*nrf_modem_at_scanf"
-	     FUNC_EXCLUDE ".*nrf_modem_at_printf")
+	     FUNC_EXCLUDE ".*nrf_modem_at_printf"
+	     WORD_EXCLUDE "__nrf_modem_(printf|scanf)_like\(.*\)")
```
Comment (trantanen) [inline tests/lib/date_time/CMakeLists.txt:18]: I don't fully understand why this is needed only in date_time tests and not location, lte_lc, lte_lc_api and sms tests which also has mock for `nrf_modem_at.h` and excludes `nrf_modem_at_printf` and `nrf_modem_at_scanf`.

Comment (trantanen) [reply tests/lib/date_time/CMakeLists.txt:18]: Hmm, looking quickly, seems those other tests are not run by twister so I'm expecting this to cause regression.

Comment (anhmolt) [reply tests/lib/date_time/CMakeLists.txt:18]: Alright, good catch.

Comment (anhmolt) [reply tests/lib/date_time/CMakeLists.txt:18]: Added

**Review by trantanen — APPROVED** [review]

**Review by jhirsi — APPROVED** [review]

**Review by eivindj-nordic — APPROVED** [review]

**Pull request merged**